test/Microsoft.ML.Tests/Scenarios/Api/Visibility.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Microsoft.ML.Runtime.Data;
+using Xunit;
+
+namespace Microsoft.ML.Tests.Scenarios.Api
+{
+    public partial class ApiScenariosTests
+    {
+        /// <summary>
+        /// Visibility: It should, possibly through the debugger, be not such a pain to actually
+        /// see what is happening to your data when you apply this or that transform. E.g.: if I
+        /// were to have the text "Help I'm a bug!" I should be able to see the steps where it is
+        /// normalized to "help i'm a bug" then tokenized into ["help", "i'm", "a", "bug"] then
+        /// mapped into term numbers [203, 25, 3, 511] then projected into the sparse
+        /// float vector {3:1, 25:1, 203:1, 511:1}, etc. etc.
+        /// </summary>
+        [Fact]
+        void Visibility()
+        {
+            var dataPath = GetDataPath(SentimentDataPath);
+            var testDataPath = GetDataPath(SentimentTestPath);
+
+            using (var env = new TlcEnvironment(seed: 1, conc: 1))
+            {
+                // Pipeline.
+                var loader = new TextLoader(env, MakeSentimentTextLoaderArgs(), new MultiFileSource(dataPath));
+
+                var trans = TextTransform.Create(env, MakeSentimentTextTransformArgs(false), loader);
+
+                using (var cursor = trans.GetRowCursor(x => true))
+                {
+                    Assert.True(cursor.Schema.TryGetColumnIndex("SentimentText", out int textColumn));
+                    Assert.True(cursor.Schema.TryGetColumnIndex("Features_TransformedText", out int transformedTextColumn));
```

Comment:
>Features_TransformedText [](start = 65, length = 24)

I'm just trying to speak from a user who is new to ML.NET and I think they will have questions about this column name, how is one supposed to know such intermediate columns generated by the text transform? what are the other intermediate columns generated? what if I already had a column with this name before text transform was applied, will this hide my column or a suffix will be added. 

Please document all this.

Reply:
See code with comment above

Reply:
Great, you are showing how to get column names, slight improvement in the example but lets focus on what you are advertising in the summary, you are saying "possibly through the debugger, be not such a pain to actually see what is happening to your data when you apply this or that transform" is this really possible with your example for cases where I have a column with name "FOO" and I apply two or more transforms to it that don't produce intermediate columns(and hide the src column) then how will I see what the first transform did? Either you show this is done or change the documentation to reflect limitations with this example.

 

---
In reply to: [210415330](https://github.com/dotnet/machinelearning/pull/653#discussion_r210415330) [](ancestors = 210415330)